metrics.go

```diff
@@ -0,0 +1,32 @@
+package caddy
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// define and register the metrics used in this package.
+func init() {
+	initAdminMetrics()
+	prometheus.MustRegister(prometheus.NewBuildInfoCollector())
+
+	// TODO: could be useful (import "github.com/povilasv/prommod")
+	// prometheus.MustRegister(prommod.NewCollector(ns))
+}
+
+// adminMetrics is a collection of metrics that can be tracked for the admin API.
+// Call initAdminMetrics to initialize.
+var adminMetrics = struct {
+	requestCount *prometheus.CounterVec
+}{}
+
+func initAdminMetrics() {
+	ns := "admin"
+	sub := "http"
+	adminMetrics.requestCount = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: ns,
+		Subsystem: sub,
+		Name:      "requests_total",
+		Help:      "Counter of requests made to admin endpoints.",
+	}, []string{"handler", "path", "code", "method"})
+}
```

Comment:
I generally recommend using `const`s for Namespace and Subsystem to provide the Go style compile-time typo detection.

Reply:
I do too, but since these are only used once -- and as part of a variable initialization -- they're effectively consts IMO. And inlining these values eliminates the need to separately initialize the variable, which is error-prone.

Reply:
> since these are only used once

They're used twice now 😉 - I added an error counter like in the `caddyhttp` metrics. I feel like tracking errors in the admin API would be quite useful too.